common/setups/rasr/hybrid_decoder.py

```diff
@@ -0,0 +1,189 @@
+__all__ = ["HybridDecoder"]
+
+from typing import Dict, List, Optional, Type, Union
+
+from sisyphus import tk
+
+import i6_core.rasr as rasr
+import i6_core.recognition as recog
+import i6_core.returnn as returnn
+
+from i6_core.returnn.flow import (
+    make_precomputed_hybrid_tf_feature_flow,
+    add_tf_flow_to_base_flow,
+)
+
+from .base_decoder import BaseDecoder
+from .config.am_config import AmRasrConfig
+from .config.lex_config import LexiconRasrConfig
+from .config.lm_config import (
+    ArpaLmRasrConfig,
+    TfRnnLmRasrConfig,
+    SimpleTfNeuralLmRasrConfig,
+    CombineLmRasrConfig,
+)
+from .util.decode import (
+    RecognitionParameters,
+    SearchJobArgs,
+    Lattice2CtmArgs,
+    ScliteScorerArgs,
+    OptimizeJobArgs,
+    PriorPath,
+)
+
+LmConfig = Union[
+    ArpaLmRasrConfig,
+    TfRnnLmRasrConfig,
+    SimpleTfNeuralLmRasrConfig,
+    CombineLmRasrConfig,
+]
+
+
+class HybridDecoder(BaseDecoder):
+    def __init__(
+        self,
+        rasr_binary_path: tk.Path,
+        rasr_arch: "str" = "linux-x86_64-standard",
+        compress: bool = False,
+        append: bool = False,
+        unbuffered: bool = False,
+        compress_after_run: bool = True,
+        search_job_class: Type[tk.Job] = recog.AdvancedTreeSearchJob,
+        scorer_job_class: Type[tk.Job] = recog.ScliteJob,
+        alias_output_prefix: str = "",
+        returnn_root: Optional[tk.Path] = None,
+        returnn_python_home: Optional[tk.Path] = None,
+        returnn_python_exe: Optional[tk.Path] = None,
+        blas_lib: Optional[tk.Path] = None,
+        search_numpy_blas: bool = True,
+        required_native_ops: Optional[List[str]] = None,
+    ):
+        """
+        Class to perform Hybrid Decoding with a RETURNN model
+
+        param rasr_binary_path: path to the rasr binary folder
+        param rasr_arch: RASR compile architecture suffix
+        """
+        super().__init__(
+            rasr_binary_path=rasr_binary_path,
+            rasr_arch=rasr_arch,
+            compress=compress,
+            append=append,
+            unbuffered=unbuffered,
+            compress_after_run=compress_after_run,
+            search_job_class=search_job_class,
+            scorer_job_class=scorer_job_class,
+            alias_output_prefix=alias_output_prefix,
+        )
+
+        self.returnn_root = returnn_root
+        self.returnn_python_home = returnn_python_home
+        self.returnn_python_exe = returnn_python_exe
+
+        self.blas_lib = blas_lib
+        self.search_numpy_blas = search_numpy_blas
+        self.required_native_ops = (
+            required_native_ops if required_native_ops is not None else []
+        )
+        self.native_ops = []
+
+    def init_decoder(
```

Comment:
as this method makes nothing special and has a very generic name, would we move it in the base class?